enhancements/operator-sdk-builds-catalogs.md

```diff
@@ -0,0 +1,179 @@
+In fact, it is the only entrypoint into the world of managing catalogs, likely for adoption reasons and
+because these processes are largely opaque and complicated, even though the spec is not.
+While `opm` is intended for "power users" (read: those who need to manipulate catalog images),
+certain subcommands like `index` and its related libraries can be used fairly easily by anyone
+`operator-sdk` should expose functionality of `opm` in a limited and sanely-defaulted way
+to enable building catalog images.
+
+### Goals
+
+- Expose `opm index add` functionality to `operator-sdk` users for all supported Operator types.
+- Use existing libraries/`opm` itself to expose the desired functionality.
+- Increase understanding of the Operator packaging lifecycle.
+
+### Non-Goals
+
+- Re-implement any part of the catalog management process.
+- Add any catalog-related code or files to `operator-sdk` projects, ex. `index.Dockerfile`.
+- Add any other functionality other than build a catalog locally (`opm index add`).
+
+## Proposal
+
+### User Stories
+
+#### Build a catalog using `opm`
+
+As an Operator developer, I want to easily build a catalog from my Operator's bundles
+without having to know the details of _how_ it is built.
+
+#### Understand what a catalog is
+
+As an Operator developer, I should understand what a catalog is and why I would consider
+building one. I also should understand how bundling and building a catalog ties into shipping my Operator.
+
+### Implementation Details/Notes/Constraints
+
+There are two ways of adding `opm index add` functionality to `operator-sdk`:
+1. Create a `catalog-build` Makefile rule that calls `opm index add`.
+1. Add a `operator-sdk catalog build` subcommand (discussed in the [alternatives](#subcommand-approach) section).
+
+The Makefile approach is by far the best for a few reasons:
+- Users have full control over which `opm` version to use.
+- Users have full control over defaults for their project, including image tags.
+- Users get experience using `opm`, but only the relevant part(s).
+  - They can easily add more `opm` targets if they want.
+
+The Makefile additions would look like:
+```make
+VERSION ?= 0.1.0
+IMAGE_TAG_BASE ?= quay.io/example/my-operator
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:$(VERSION)
+
+bundle-push:
+	$(MAKE) docker-push IMG=$(BUNDLE_IMG)
+
+OPM = ./bin/opm
+opm:
+  # Download `opm` from github.com/operator-framework/operator-registry/releases
+
+BUNDLE_IMGS ?= $(BUNDLE_IMG)
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(VERSION)
+catalog-build: opm
+	$(OPM) index add --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS)
+
+catalog-push:
+	$(MAKE) docker-push IMG=$(CATALOG_IMG)
+```
+
+In addition, consider these changes to the current Makefile:
+
+```diff
+VERSION ?= 0.1.0
++IMAGE_TAG_BASE ?= quay.io/example/my-operator
+-BUNDLE_IMG ?= controller-bundle:$(VERSION)
++BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
```

Comment:
I like this change, but would it be considered a breaking change?

Reply:
Nope, this is just a convenience thing.